src/main/java/com/team766/hal/GenericRobotMain3.java

```diff
@@ -0,0 +1,174 @@
+package com.team766.hal;
+
+import com.team766.framework3.AutonomousMode;
+import com.team766.framework3.RuleEngine;
+import com.team766.framework3.SchedulerMonitor;
+import com.team766.framework3.SchedulerUtils;
+import com.team766.library.RateLimiter;
+import com.team766.logging.Category;
+import com.team766.logging.Logger;
+import com.team766.logging.Severity;
+import com.team766.web.AutonomousSelector;
+import com.team766.web.ConfigUI;
+import com.team766.web.Dashboard;
+import com.team766.web.DriverInterface;
+import com.team766.web.LogViewer;
+import com.team766.web.ReadLogs;
+import com.team766.web.WebServer;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+
+// Team 766 - Robot Interface Base class
+
+public final class GenericRobotMain3 implements GenericRobotMain {
+    private RobotConfigurator3 configurator;
+    private RuleEngine m_oi;
+    private RuleEngine m_lights;
+
+    private WebServer m_webServer;
+    private AutonomousSelector<AutonomousMode> m_autonSelector;
+    private AutonomousMode m_autonMode = null;
+    private Command m_autonomous = null;
+
+    // Reset the autonomous routine if the robot is disabled for more than this
+    // number of seconds.
+    private static final double RESET_IN_DISABLED_PERIOD = 10.0;
+    private double m_disabledModeStartTime;
+
+    private RateLimiter m_lightUpdateLimiter = new RateLimiter(0.05);
+
+    private boolean faultInRobotInit = false;
+    private boolean faultInAutoInit = false;
+    private boolean faultInTeleopInit = false;
+
+    public GenericRobotMain3(RobotConfigurator3 configurator) {
+        SchedulerUtils.reset();
+        SchedulerMonitor.start();
+
+        this.configurator = configurator;
+        m_autonSelector = new AutonomousSelector<>(configurator.getAutonomousModes());
+        m_webServer = new WebServer();
```

Comment:
I was trying to avoid `GenericGenericRobotMain` 🙃 

It felt like some short-term maintenance burden from a little duplicated code in the two `GenericRobotMain`s was fine, rather than an over-engineered solution.